PublicAPI/Microsoft.ApplicationInsights.dll/net45/PublicAPI.Unshipped.txt

```diff
@@ -33,5 +33,237 @@ Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.IHeartbeatPro
+override Microsoft.ApplicationInsights.Metrics.MetricIdentifier.GetHashCode() -> int
+override Microsoft.ApplicationInsights.Metrics.MetricIdentifier.ToString() -> string
+override Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.Equals(object obj) -> bool
+override Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.GetHashCode() -> int
+static Microsoft.ApplicationInsights.MetricConfigurationsExtensions.Measurement(this Microsoft.ApplicationInsights.MetricConfigurations metricConfigPresets) -> Microsoft.ApplicationInsights.Metrics.MetricConfigurationForMeasurement
+static Microsoft.ApplicationInsights.MetricConfigurationsExtensions.SetDefaultForMeasurement(this Microsoft.ApplicationInsights.MetricConfigurations metricConfigPresets, Microsoft.ApplicationInsights.Metrics.MetricConfigurationForMeasurement defaultConfigurationForMeasurement) -> void
+static Microsoft.ApplicationInsights.Metrics.Extensibility.MetricConfigurationExtensions.Constants(this Microsoft.ApplicationInsights.Metrics.MetricConfigurationForMeasurement measurementConfig) -> Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.AggregateKindConstants
+static Microsoft.ApplicationInsights.Metrics.Extensibility.MetricConfigurationExtensions.Constants(this Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement measurementConfig) -> Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.AggregateKindConstants
+static Microsoft.ApplicationInsights.Metrics.Extensibility.MetricExtensions.GetConfiguration(this Microsoft.ApplicationInsights.Metric metric) -> Microsoft.ApplicationInsights.Metrics.MetricConfiguration
+static Microsoft.ApplicationInsights.Metrics.Extensibility.MetricSeriesExtensions.GetConfiguration(this Microsoft.ApplicationInsights.Metrics.MetricSeries metricSeries) -> Microsoft.ApplicationInsights.Metrics.IMetricSeriesConfiguration
+static Microsoft.ApplicationInsights.Metrics.Extensibility.TelemetryClientExtensions.GetMetricManager(this Microsoft.ApplicationInsights.TelemetryClient telemetryClient, Microsoft.ApplicationInsights.MetricAggregationScope aggregationScope) -> Microsoft.ApplicationInsights.Metrics.MetricManager
+static Microsoft.ApplicationInsights.Metrics.MetricIdentifier.DefaultMetricNamespace.get -> string
+static Microsoft.ApplicationInsights.Metrics.MetricIdentifier.DefaultMetricNamespace.set -> void
+static Microsoft.ApplicationInsights.Metrics.TelemetryConfigurationExtensions.GetMetricManager(this Microsoft.ApplicationInsights.Extensibility.TelemetryConfiguration telemetryPipeline) -> Microsoft.ApplicationInsights.Metrics.MetricManager
+static readonly Microsoft.ApplicationInsights.MetricConfigurations.Common -> Microsoft.ApplicationInsights.MetricConfigurations
+virtual Microsoft.ApplicationInsights.Metrics.MetricConfiguration.Equals(Microsoft.ApplicationInsights.Metrics.MetricConfiguration other) -> bool
```

Comment:
`MetricConfiguration` namespace looks empty - only two methods?

Reply:
> MetricConfiguration namespace looks empty - only two methods?

I am not sure what you mean by this comment. Please let me make some educated guesses about what you mean and try to provide as much information as possible.

* There is no namespace `MetricConfiguration`. There is a type with this name, but it has more than two members. I have described the usage and the purpose of that class in great details just a bit above in the response to your comment about `MetricIdentifier`.

* There is also an identifier with a similar name `MetricConfigurations`. Is is also not a namespace, but a static class. Its usage and purpose is described in the doc comments and in examples 1, 2, 3 and 5:
https://github.com/Microsoft/ApplicationInsights-dotnet/blob/feature/PreAggMetrics.Master/Test/Microsoft.ApplicationInsights.Test/Shared/Metrics/MetricsExamples.cs

Please let me provide additional details here.
`MetricConfigurations` represents an easily discoverable (via Intellisense) point for pre-defined metric configurations:

The common `TelemetryClient.GetMetric(..)` method has several overloads expecting a parameter of type `MetricConfiguration`. When attempting to type this term, the user discovers the static instance `MetricConfigurations.Common` (in the same namespace). We use extension methods to define common metric configurations:
In this version, the base SDK defines `MetricConfigurations.Common.Measurement()`.
The *Metrics.Extensions* package is planned to add:
- `MetricConfigurations.Common.Accumulator()`
- `MetricConfigurations.Common.Gauge()`
- `MetricConfigurations.Common.NaiveDistinctCount()`

You have seen the complete ready and tested code for this additional configurations and you requested that we move it to a different repo. It is currently available in the SDK labs Repo.
Users may also add their own common configurations via the same extension method mechanism.

Please let me know if this answers your question.
If you require adjustments to these APIs, please request *precisely* the changes you would like to see so that I can make them.
Thank you.